white-paper/wp.tex

```diff
@@ -65,34 +65,177 @@
+\todo{Analyze this tool}
+
+  \item[\href{http://www.oversim.org/wiki}{\textbf{OverSim}}]
+Open source, GNU AGPLv3. \href{http://www.oversim.org/wiki/OverSimDownload}{Code in archive here.}
+
+OverSim is an open-source overlay and peer-to-peer network simulation framework for the \href{https://omnetpp.org/}{OMNeT++} simulation environment. The simulator contains several models for structured (e.g. Chord, Kademlia, Pastry) and unstructured (e.g. GIA) P2P systems and overlay protocols.\par
+This is the addon over OMNeT++ engine, with possibilities for customization. Code is opened, but not in github. \todo{Analyze possibilities of modification}
+\todo{analyze the source code}
+However it can be useful as library with \href{https://en.wikipedia.org/wiki/Kademlia}{Kademlia} implementation.
+
+  \item[\href{https://www.anylogic.com/}{\textbf{AnyLogic}}]
+Designed as most complete solution for any case, and allows to use all three modern methods:
+    \begin{itemize}
+        \item Discrete Event
+        \item Agent-Based
+        \item System Dynamics
+    \end{itemize}
+    \begin{quote}
+    The three methods can be used in any combination, with one software, to simulate business systems of any complexity. A various visual modeling languages can be used: process flowcharts, state charts, action charts, and stock and flow diagrams. AnyLogic provides industry-specific libraries.\par
+    \end{quote}
+It's a proprietary software that can't be used for free (except education propose) and the source code is closed. Moreover, this product is popular in industry as many others and not aimed to modelling the network infrastructure.
+
+\end{description}
+
+\section{Architecture}
+\label{sec:arch}
+
+Simulator creates a graph of physical nodes where placed distributed github repositories. Every repository is placed on some number (depending on settings) of physical nodes.
+
+All nodes for every repository must be connected, but not mandatory directly. Update the repository on one node must 
+
+System must have an Event-Engine to simulate random environment processes and emulate payload (from mocked RPC requests of DeGitX)
+
+System contents the following modules:
+\begin{description}
+    \item[Graph Generator.]
+    Based on predefined settings, module should create bi-directional graph. Nodes are the physical servers which contain one or more repositories. Each node has own parameters, which must be set based on settings with some minor random delta:
+    \begin{itemize}
+        \item Read operation speed    (MB/sec)
+        \item Write delay (Mb/sec)
+        \item State (Active/Down)
+    \end{itemize}
+    Edges of graph are routes from one Node to another. They must have their own configuration properties:
+    \begin{itemize}
+        \item Throughput (max Mb/sec)
+        \item State (Active/Down)
+    \end{itemize}
+    Parameters for nodes and edges must be set with given distribution method. (Normal, random, etc)
+    
+    \item[Repository.]
```

Comment:
@IlyaMoskva again, I'd call it node state

Reply:
@g4s8 , I agree